modules/apps/foundation/user-associated-data/user-associated-data-web/src/main/java/com/liferay/user/associated/data/web/internal/portlet/action/ManageUserAssociatedDataSummaryMVCRenderCommand.java

```diff
@@ -42,7 +49,51 @@ public String render(
 			RenderRequest renderRequest, RenderResponse renderResponse)
 		throws PortletException {
 
+		try {
+			User selUser = PortalUtil.getSelectedUser(renderRequest);
+
+			int step = _determineStep(selUser);
+
+			renderRequest.setAttribute("step", step);
```

Comment:
Two thoughts on this one:
1. If we keep it as an attribute, we should make sure to add the attribute name as a constant in a `*WebKeys*` file in this module.
2. The other approach we can take is creating a `*DisplayContext.java` class for this view.  It would be constructed on the jsp (accepting the request and response), then would have a method called `getStep` where you would put all the logic below.  See usages of `UserDisplayContext.java` as an example.